analysis/2020-11-09-anapaulagomes-propostas-dos-candidatos-a-prefeito.ipynb

```diff
@@ -0,0 +1,252 @@
+    "\n",
+    "Para rodar essa análise você deve:\n",
+    "* Baixar o arquivo [propostas.7z](https://github.com/augusto-herrmann/eleicoes-2020-planos-de-governo/blob/main/dados/propostas.7z)\n",
+    "* Extraí-lo na pasta `analysis` ou modificar a variável `proposals_directory` com o caminho completo para a pasta\n",
+    "\n",
+    "Padrão de pastas esperado: `propostas > sigla do estado > cidades`\n",
+    "\n",
+    "---\n",
+    "\n",
+    "Nessa análise vamos visualizar uma nuvem de palavras das propostas a\n",
+    "candidatos de toda Bahia e de Feira de Santana.\n",
+    "\n",
+    "Com isso queremos responder as perguntas:\n",
+    "\n",
+    "1. Quais são as palavras de destaque entre os candidatos a prefeito de Feira?\n",
+    "2. Quais são as palavras de destaque entre todos os candidatos a prefeito de Feira juntos?\n",
+    "3. Quais são as palavras de destaque entre todos os candidatos a prefeito da Bahia?"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "\n",
+    "\n",
+    "target_state = \"ba\"  # sigla do estado em minúsculo\n",
+    "target_city = \"feira-de-santana\"  # nome da cidade separado por hífens\n",
+    "proposals_directory = \"propostas/\"\n",
+    "proposals_and_cities = {}\n",
+    "\n",
+    "# estrutura: estado > cidade > proposta-<candidato>.txt\n",
+    "for folder, _, files in os.walk(proposals_directory):\n",
+    "    if files:\n",
+    "        index_city = folder.rfind(\"/\")\n",
+    "        city = folder[index_city+1:]\n",
+    "        index_state = folder[:index_city].rfind(\"/\")\n",
+    "        state = folder[index_state+1:index_city]\n",
+    "        proposals_and_cities[f\"{state}/{city}\"] = files\n",
+    "proposals_and_cities[f\"{target_state}/{target_city}\"]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Propostas por candidato"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from scripts.nlp import remove_portuguese_stopwords\n",
+    "\n",
+    "import pandas as pd\n",
+    "import texthero as hero\n",
+    "from texthero import preprocessing\n",
+    "\n",
+    "\n",
+    "custom_pipeline = [preprocessing.remove_digits,\n",
+    "                   preprocessing.remove_diacritics,\n",
+    "                   preprocessing.remove_whitespace]\n",
+    "\n",
+    "candidates_and_proposals = {}\n",
+    "for file_name in proposals_and_cities[f\"{target_state}/{target_city}\"]:\n",
+    "    candidate_name = file_name[12:]\n",
+    "    candidate_name = candidate_name.replace(\".txt\", \"\")\n",
+    "    candidate_name = candidate_name.replace(\"-\", \" \")\n",
+    "    candidate_name = candidate_name.title()\n",
+    "    print(candidate_name)\n",
+    "    content = open(f\"{directory}/{file_name}\").read()\n",
```

Comment:
Deu erro nessa linha:

`Name 'directory' is not defined`

Reply:
Aqui rodou 

`content = open(f"{proposals_directory}{target_state}/{target_city}/{file_name}").read()`

O notebook tá na pasta `analysis`. Essa pasta tb contém a pasta `propostas`. E aí tem propostas > BA > feira-de-santana, do jeito que tá indicado na linha 20

Reply:
Engraçado, o meu programa de descompactação descompactou com uma estrutura diferente. 👍🏽